tests/test_api_machines_serviceid.py

```diff
@@ -0,0 +1,310 @@
+"""
+This testcase is used to test the REST API  in api/machines.py
+to fetch machine information and to attach token to machines
+"""
+import passlib
+
+from privacyidea.lib.user import User
+from .base import MyApiTestCase
+import json
+from privacyidea.lib.token import init_token, get_tokens, remove_token
+from privacyidea.lib.machine import attach_token, detach_token, ANY_MACHINE, NO_RESOLVER
+from privacyidea.lib.policy import (set_policy, delete_policy, ACTION, SCOPE)
+
+HOSTSFILE = "tests/testdata/hosts"
+
+SSHKEY = "ssh-rsa " \
+         "AAAAB3NzaC1yc2EAAAADAQABAAACAQDJy0rLoxqc8SsY8DVAFijMsQyCv" \
+         "hBu4K40hdZOacXK4O6OgnacnSKN56MP6pzz2+4svzvDzwvkFsvf34pbsgD" \
+         "F67PPSCsimmjEQjf0UfamBKh0cl181CbPYsph3UTBOCgHh3FFDXBduPK4DQz" \
+         "EVQpmqe80h+lsvQ81qPYagbRW6fpd0uWn9H7a/qiLQZsiKLL07HGB+NwWue4os" \
+         "0r9s4qxeG76K6QM7nZKyC0KRAz7CjAf+0X7YzCOu2pzyxVdj/T+KArFcMmq8V" \
+         "dz24mhcFFXTzU3wveas1A9rwamYWB+Spuohh/OrK3wDsrryStKQv7yofgnPMs" \
+         "TdaL7XxyQVPCmh2jVl5ro9BPIjTXsre9EUxZYFVr3EIECRDNWy3xEnUHk7Rzs" \
+         "734Rp6XxGSzcSLSju8/MBzUVe35iXfXDRcqTcoA0700pIb1ANYrPUO8Up05v4" \
+         "EjIyBeU61b4ilJ3PNcEVld6FHwP3Z7F068ef4DXEC/d7pibrp4Up61WYQIXV/" \
+         "utDt3NDg/Zf3iqoYcJNM/zIZx2j1kQQwqtnbGqxJMrL6LtClmeWteR4420uZx" \
+         "afLE9AtAL4nnMPuubC87L0wJ88un9teza/N02KJMHy01Yz3iJKt3Ou9eV6kqO" \
+         "ei3kvLs5dXmriTHp6g9whtnN6/Liv9SzZPJTs8YfThi34Wccrw== " \
+         "NetKnights GmbH"
+SSHKEY_ecdsa = "ecdsa-sha2-nistp256 AAAAE2VjZHNhLXNoYTItbmlzdHAyNTYAAAAIbmlzd" \
+               "HAyNTYAAABBBHGCdIk0pO1HFr/mF4oLb43ZRyQJ4K7ICLrAhAiQERVa0tUvyY5TE" \
+               "zurWTqxSMx203rY77t6xnHLZBMPPpv8rk0= cornelius@puck"
+OTPKEY = "3132333435363738393031323334353637383930"
+
+
+class APIMachinesServiceIDTestCase(MyApiTestCase):
+
+    serial1 = "SSHKEY1"
+    serial2 = "SSHKEY2"
+    serviceID1 = "webserver"
+    serviceID2 = "mailserver"
+
+    def test_01_create_sshkeys(self):
+        # create two tokens
+        token_obj = init_token({"serial": self.serial1, "type": "sshkey",
+                                "sshkey": SSHKEY})
+        self.assertEqual(token_obj.type, "sshkey")
+        token_obj = init_token({"serial": self.serial2, "type": "sshkey",
+                                "sshkey": SSHKEY_ecdsa})
+        self.assertEqual(token_obj.type, "sshkey")
+
+    def test_02_attach_tokens(self):
+        # Attach token S1 to webserver and mailserver
```

Comment:
I would only write here **_"# Attach token S1 to webserver"_** since each step is described below.

Reply:
My point was, to give an overview of the token attachements at the beginning of the test, so that it is clear what can be checked later.
I am doing a bit reformatting

Reply:
Done in f4469ac